agent/consul/state/acl.go

```diff
@@ -560,6 +561,68 @@ func (s *Store) CanBootstrapACLToken() (bool, uint64, error) {
 	return false, out.(*IndexEntry).Value, nil
 }
 
+func (s *Store) resolveACLLinks(tx *memdb.Txn, links []agentpb.ACLLink, getName func(*memdb.Txn, string) (string, error)) (int, error) {
+	var numValid int
+	for linkIndex, link := range links {
+		if link.ID != "" {
+			name, err := getName(tx, link.ID)
+
+			if err != nil {
+				return 0, err
+			}
+
+			// the name doesn't matter here
+			if name != "" {
+				links[linkIndex].Name = name
+				numValid++
+			}
```

Comment:
Should this have a branch for `} else if !allowMissing {` like some of the existing blocks that look similar to this?

Reply:
The `getName` func provided is supposed to handle that case and return an error if thats the desired behavior. I am adding some comments about that and both this and fixupACLLinks intended usage and behavior.